### PR TITLE
HDDS-6382. EC: ListPipelines command should consider EC Containers

### PR DESCRIPTION
**Pull request by sodonnel**
## What changes were proposed in this pull request?

In ListPipeline.java we have:

```
@Override
public void execute(ScmClient scmClient) throws IOException {
  Stream<Pipeline> stream = scmClient.listPipelines().stream();
  if (!Strings.isNullOrEmpty(factor)) {
    stream = stream.filter(
        p -> ReplicationConfig.getLegacyFactor(p.getReplicationConfig())
            .toString().compareToIgnoreCase(factor) == 0);
  }
  if (!Strings.isNullOrEmpty(state)) {
    stream = stream.filter(p -> p.getPipelineState().toString()
        .compareToIgnoreCase(state) == 0);
  }
  stream.forEach(System.out::println);
} 
```

This code will fail if the pipeline is an EC pipeline. We should also allow for filtering on EC pipelines.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-6382

## How was this patch tested?

New unit tests and verified in docker cluster

### CODE REVIEW

File: hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/pipeline/ListPipelinesSubcommand.java [hunk 1]
```diff
@@ -39,23 +41,41 @@
...
-  @CommandLine.Option(names = {"-ffc", "--filterByFactor"},
-      description = "Filter listed pipelines by Factor(ONE/one)",
+  @CommandLine.Option(names = {"-t", "--type"},
+      description = "Filter listed pipelines by replication type, RATIS or EC",
       defaultValue = "")
-  private String factor;
+  private String replicationType;
 
-  @CommandLine.Option(names = {"-fst", "--filterByState"},
-      description = "Filter listed pipelines by State(OPEN/CLOSE)",
```
Comment (adoroszlai) [inline hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/pipeline/ListPipelinesSubcommand.java:48]: Can we keep these old options for backward compatibility?

Comment (sodonnel) [reply hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/pipeline/ListPipelinesSubcommand.java:48]: I can add them back in, but in my opinion we shouldn't. The current options don't fit with the convention of type / replication used everywhere else and why do we even have "ffc" for "filter by factor"? 

I am all for compatibility on important areas, but this is for a likely rarely used command that is not critical for any operations on the cluster. The extra options will make the help text look messy and we will end up carrying around 4 ways to pass the factor forever.

**Review by umamaheswararao — APPROVED** [review]
LGTM

Thanks Attila for reviews.

File: hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/pipeline/ListPipelinesSubcommand.java (hunk 1 above)
Comment (adoroszlai) [reply hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/pipeline/ListPipelinesSubcommand.java:48]: Thanks for restoring them.  This command may look like rarely used, but it is useful for automated testing (validating cluster state).

I agree that these old options are non-standard (and clumsy), and would like to get rid of them eventually.  But we should deprecate them first.  Co-existing old and new options allow non-breaking upgrade, giving time to update test scripts.  If we simply replace them, downstream users need to adapt at the same time as bringing in these Ozone changes, which can be a chicken-egg problem.

**Pull request merged**